packages/react-native-ui-lib/src/components/floatingButton/index.tsx

```diff
@@ -191,21 +212,24 @@ class FloatingButton extends PureComponent<FloatingButtonProps> {
       return false;
     }
 
-    return (
-      <View
-        row={this.isSecondaryHorizontal}
-        center={this.isSecondaryHorizontal || !fullWidth}
-        pointerEvents="box-none"
-        animated
-        style={[styles.container, this.getAnimatedStyle()]}
-        testID={testID}
-      >
-        {this.renderOverlay()}
-        {this.isSecondaryHorizontal && this.renderSecondaryButton()}
-        {this.renderButton()}
-        {this.isSecondaryVertical && this.renderSecondaryButton()}
-      </View>
-    );
+    if (button || secondaryButton) {
+      const shouldUseHorizontalLayout = this.isSecondaryHorizontal || this.isSecondaryOnly;
+      return (
+        <View
+          row={shouldUseHorizontalLayout}
+          center={shouldUseHorizontalLayout || !fullWidth}
+          pointerEvents="box-none"
+          animated
+          style={[styles.container, this.getAnimatedStyle()]}
+          testID={testID}
+        >
+          {this.renderOverlay()}
+          {shouldUseHorizontalLayout && this.renderSecondaryButton()}
+          {button && this.renderButton()}
+          {this.isSecondaryVertical && this.renderSecondaryButton()}
+        </View>
```

Comment:
This is getting less readable IMO.
WDYT about splitting it to horizontal vs vertical?